xarray/core/indexes.py

```diff
@@ -575,13 +578,19 @@ class PandasIndex(Index):
 
     __slots__ = ("index", "dim", "coord_dtype")
 
-    def __init__(self, array: Any, dim: Hashable, coord_dtype: Any = None):
-        # make a shallow copy: cheap and because the index name may be updated
-        # here or in other constructors (cannot use pd.Index.rename as this
-        # constructor is also called from PandasMultiIndex)
-        index = safe_cast_to_index(array).copy()
+    def __init__(
+        self, array: Any, dim: Hashable, coord_dtype: Any = None, *, fastpath=False
+    ):
```

Comment:
I'm a bit wary of adding `fastpath` everywhere. It makes things a bit hard to reason about. Is this one that consequential? We do first check for `isinstance(pd.Index)` in `safe_cast_to_index`...

Reply:
Anecdotally, I remember thinking that indexes are created and copied a few too many times. I believe this was done for correctness in a big refactor, but reducing any useless copies would be a big win since we would call `_replace` less.

Reply:
> Anecdotally, I remember thinking that indexes are created and copied a few too many times. 

this in general is my feeling after my repeated deep diving into the codebase.